chess/4-database/starter-code/serverTests/PersistenceTests.java

```diff
@@ -56,10 +58,14 @@ public void persistenceTest() throws TestException {
         joinRequest.playerColor = ChessGame.TeamColor.WHITE;
         serverFacade.verifyJoinPlayer(joinRequest, auth);
 
+        validateDatabase(3, "Database not populate after game play");
```

Comment:
This aligns with the concerns that I have about the implementation. Your suggestion does make it handle the edge cases better.

I need to rewrite the query because Dr. Wilkerson is experiencing some strange behavior. When I do that I will integrate the spirit of your suggestions in the final commit.

I think we can just make a utility method that gets the row count for the current database and let the individual test validate it's assumptions about what the row count should be.